robotidy/files.py

```diff
@@ -84,12 +84,19 @@ def get_gitignore(root: Path) -> PathSpec:
     return PathSpec.from_lines("gitwildmatch", lines)
 
 
-def path_is_excluded(
-    normalized_path: str,
-    pattern: Optional[Pattern[str]],
+def should_parse_path(
+    path: Path, exclude: Optional[Pattern[str]], extend_exclude: Optional[Pattern[str]], gitignore: Optional[PathSpec]
 ) -> bool:
-    match = pattern.search(normalized_path) if pattern else None
-    return bool(match and match.group(0))
+    normalized_path = str(path)
+    for pattern in (exclude, extend_exclude):
+        match = pattern.search(normalized_path) if pattern else None
+        if bool(match and match.group(0)):
+            return False
+    return (
+        not (gitignore is not None and gitignore.match_file(path))
+        and not (path.is_file() and path.suffix not in INCLUDE_EXT)
+        and not (path.is_dir() and exclude and exclude.match(path.name))
+    )
```

Comment:
To save a tiny bit of time it could be:
```
if gitignore is not None and gitignore.match_file(path):
    return False
if path.is_file():
    return path.suffix in INCLUDE_EXT
else:
    if exclude and exclude.match(path.name):
        return False
    return True
```
but it's bit more verbose, so it's up to you - I will approve PR.

Reply:
yeah thats much more readable too lol